src/xcode/ENA/ENA/Source/Services/DSCList/DSCListProvider.swift

```diff
@@ -0,0 +1,87 @@
+//
+// 🦠 Corona-Warn-App
+//
+
+import UIKit
+import OpenCombine
+import HealthCertificateToolkit
+
+final class DSCListProvider: DSCListProviding {
+
+	// MARK: - Init
+
+	init(
+		client: DSCListFetching,
+		store: Store,
+		interval: TimeInterval = 12 * 60 * 60  // 12 hours
+	) {
+		self.client = client
+		self.store = store
+		self.interval = interval
+		// load last knows | default DSCList data
+		self.metaData = Self.loadDSCListMetaDataIfAvailable(store: store, interval: interval)
+		self.signingCertificates = CurrentValueSubject<[DCCSigningCertificate], Never>(metaData.signingCertificate)
+
+		// trigger an update immediately
+		NotificationCenter.default.addObserver(self, selector: #selector(updateListIfNeeded), name: UIApplication.willEnterForegroundNotification, object: nil)
```

Comment:
To have this working, an instance of `DSCListProvider` needs to be held directly or indirectly by the app delegate.

Reply:
yes this will be done in a later PR when we start using the provider.